src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -1028,6 +1094,63 @@ ActStatus ActionHandlers::send_init_ctxt_req_to_ue(SM::ControlBlock& cb)
 	memcpy(&(icr_msg.gtp_teid), &(bearerCtxt->getS1uSgwUserFteid().fteid_m), sizeof(struct fteid));
 	memcpy(&(icr_msg.sec_key), &((ue_ctxt->getUeSecInfo().secinfo_m).kenb_key),
 			KENB_SIZE);	
+
+	bool nsa_enabled = false;
+    
+	// If UE supports DCNR
+	if (ue_ctxt->getUeNetCapab().ue_net_capab_m.u.bits.dcnr)
+	{
+		// But the network does not allow DCNR for the UE
+		if (ue_ctxt->getDcnrCapable() == false)
+		{
+			eps_nw_feature_supp_presence = true;
+
+			icr_msg.ho_restrict_list_presence = true;
+			memcpy(&(icr_msg.ho_restrict_list.serving_plmn),
+					&(ue_ctxt->getTai().tai_m.plmn_id), sizeof(struct PLMN));
```

Comment:
PLMN structure has multiple extra fields. and that causes PLMN struct to return 6-7 bytes. I guess you just want to copy mcc/mnc. In that case just pass 3 as argument to memcpy.

Reply:
Done.